src/ng/rootScope.js

```diff
@@ -1125,14 +1125,58 @@ function $RootScopeProvider() {
 
         var self = this;
         return function() {
-          var indexOfListener = namedListeners.indexOf(listener);
-          if (indexOfListener !== -1) {
-            namedListeners[indexOfListener] = null;
-            decrementListenerCount(self, 1, name);
-          }
+          deregisterListener(self, name, listener);
         };
       },
 
+      /**
+       * @ngdoc method
+       * @name $rootScope.Scope#$once
+       * @kind function
+       *
+       * @description
+       * Listens on an event of a given type at most once. The listener will be deregistered
+       * after being called the first time. See {@link ng.$rootScope.Scope#$emit $emit}
+       * for discussion of event life cycle.
+       *
+       * The event listener function format is: `function(event, args...)`. The `event` object
+       * passed into the listener has the following attributes:
+       *
+       *   - `targetScope` - `{Scope}`: the scope on which the event was `$emit`-ed or
+       *     `$broadcast`-ed.
+       *   - `currentScope` - `{Scope}`: the scope that is currently handling the event. Once the
+       *     event propagates through the scope hierarchy, this property is set to null.
+       *   - `name` - `{string}`: name of the event.
+       *   - `stopPropagation` - `{function=}`: calling `stopPropagation` function will cancel
+       *     further event propagation (available only for events that were `$emit`-ed).
+       *   - `preventDefault` - `{function}`: calling `preventDefault` sets `defaultPrevented` flag
+       *     to true.
+       *   - `defaultPrevented` - `{boolean}`: true if `preventDefault` was called.
+       *
+       * @param {string} name Event name to listen on.
+       * @param {function(event, ...args)} listener Function to call when the event is emitted.
+       */
+      $once: function(name, listener) {
```

Comment:
I think that what @Narretz is trying to say is that `$once` should be implemented as

```js
// Untested code
$once: function(name listener) {
  var delegate = this.$on(name, function() {
    delegate();
    listener.apply(null, arguments);
  });
},
```

and leave `$on` unmodified.